recipes/sambamba/meta.yaml

```diff
@@ -1,22 +1,38 @@
-{% set version='0.6.6' %}
+{% set version='0.6.7' %}
+{% set revision='895512e7' %}
+
 package:
   name: sambamba
   version: {{ version }}
 
+build:
+  number: 0
+
+extra:
+  skip-lints:
+    - uses_git_url
+
 source:
-  url: https://github.com/lomereiter/sambamba/releases/download/v{{ version }}/sambamba_v{{ version }}_linux.tar.bz2 # [linux]
-  md5: 6fc3b70ac32441f3a9c6aa87743c8e3c # [linux]
-  url: https://github.com/lomereiter/sambamba/releases/download/v{{ version }}/sambamba_v{{ version }}_osx.tar.bz2 # [osx]
-  md5: 07d82a2102126565e6796658face7c71 # [osx]
+  git_url: https://github.com/biod/sambamba
+  git_rev: {{ revision }}
+  md5: unused
 
-build:
-  number: 2
+requirements:
+  build:
+    - ldc
+    - bzip2 {{ CONDA_BZIP2 }}*
+    - zlib {{ CONDA_ZLIB }}*
+    - xz {{ CONDA_XZ }}*
```

Comment:
This jinja2-syntax was a bioconda-specific way to pin dependencies which is now outdated with the switch to conda-build 3. You can simply use the names of the libs here and in run, and pinning will happen automatically.

Moreover, use `host`  instead of build, and add the compiler to the build section. See the bwa recipe for an example: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/bwa/meta.yaml

Reply:
Thanks a lot Johannes, much appreciated! :)

I set the compiler to `ldc` but the build failed: https://circleci.com/gh/bioconda/bioconda-recipes/20195 ... I'll try now with just compiler('c') since I believe there's the C bits (lz4, htslib, etc...) that might require that directive (guessing on the internals of conda-build 3)...

Reply:
@johanneskoester, After removing the pinning, zlib doesn't seem to be found automatically:

https://circleci.com/gh/bioconda/bioconda-recipes/20201

```bash
cd lz4/lib && gcc -O3 -c lz4.c lz4hc.c lz4frame.c xxhash.c && ar rcs liblz4.a lz4.o lz4hc.o lz4frame.o xxhash.o
cd htslib && make
make[1]: Entering directory `/opt/conda/conda-bld/sambamba_1533083071240/work/htslib'
gcc -g -Wall -O2 -D_GNU_SOURCE -I.  -c -o kfunc.o kfunc.c
gcc -g -Wall -O2 -D_GNU_SOURCE -I.  -c -o knetfile.o knetfile.c
gcc -g -Wall -O2 -D_GNU_SOURCE -I.  -c -o kstring.o kstring.c
echo '/* Empty config.h generated by Makefile */' > config.h
gcc -g -Wall -O2 -D_GNU_SOURCE -I.  -c -o bgzf.o bgzf.c
In file included from bgzf.c:39:0:
htslib/bgzf.h:34:18: fatal error: zlib.h: No such file or directory
 #include <zlib.h>
                  ^
compilation terminated.
make[1]: *** [bgzf.o] Error 1
make[1]: Leaving directory `/opt/conda/conda-bld/sambamba_1533083071240/work/htslib'
make: *** [htslib-static] Error 2
```